AirCasting/BluetoothSession/UserTriggeredReconnectionController.swift

```diff
@@ -0,0 +1,67 @@
+// Created by Lunar on 24/11/2022.
+//
+
+import Foundation
+import Resolver
+
+enum UserTroggeredReconnectionError: Error {
+    case anotherActiveSessionInProgress
+    case failedToDiscoverCharacteristics
+    case failedToConnect
+    case deviceNotDiscovered
+    case airbeamConfigurationFailure
```

Comment:
I think we don't really need that detailed cases here. In my opinion we can just use 3: `activeSessionInProgress` + `failedToConnect` + `deviceUnavailable`. Generally each layer usually boils down the errors to match its domain. Here we're pretty high up, so we don't really care if reconnection failed because BT failed to discover characteristics or because the configuration failed. Those are both connection errors and can be merged at this point (at the `Controller` level the errors should be thought of as user-facing ones. We might not display them ofc, it's just my own rule of thumb)